clients/wallet/src/horizon.rs

```diff
@@ -427,30 +435,77 @@ impl HorizonClient for reqwest::Client {
 		transaction_envelope: TransactionEnvelope,
 		is_public_network: bool,
 	) -> Result<TransactionResponse, Error> {
+		let seq_no = transaction_envelope.sequence_number();
 		let transaction_xdr = transaction_envelope.to_base64_xdr();
 		let transaction_xdr = std::str::from_utf8(&transaction_xdr).map_err(Error::Utf8Error)?;
 
-		let base_url = horizon_url(is_public_network);
-		let url = format!("{}/transactions", base_url);
-
 		let params = [("tx", &transaction_xdr)];
-		let response =
-			self.post(url).form(&params).send().await.map_err(Error::HttpFetchingError)?;
-
-		interpret_response::<TransactionResponse>(response).await.map_err(|e| match e {
-			Error::HorizonSubmissionError { title, status, reason, envelope_xdr }
-				if envelope_xdr.is_none() =>
-			{
-				// let's add a transaction envelope, if possible
-				Error::HorizonSubmissionError {
-					title,
-					status,
-					reason,
-					envelope_xdr: Some(transaction_xdr.to_string()),
-				}
-			},
-			other => other,
-		})
+
+		let mut server_error_count = 3;
+		loop {
```

Comment:
My opinion that It is better to move value '3' to configuration instead of hardcoding.

Reply:
@RustNinja as I do not know what our specific num of attempts should be, i added a method of StellarWallet:
**_`pub fn with_max_retry_attempt(mut self, max_retries: u8) -> Self {...`_**

and we can configure it in system.rs.